src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,213 @@
+
+        assertThat(defaultController.annotations).anySatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}}
+
+        assertThat(defaultController.parameters).anySatisfy{parameter ->
+            assertThat(parameter.name).isEqualTo("authentication")
+            assertThat(parameter.type.toString()).isEqualTo("io.micronaut.security.authentication.Authentication")
+        }
+
+
+    }
+
+    @Test
+    fun `ensure that only operation-level security being defined will result in the correctly generated code`() {
+        val controller = setupTest("operation_authentication_without_global.yml")
+
+        val prohibitedController = (controller.find{it.name == "ProhibitedController"}!!.members.single() as TypeSpec).funSpecs.single()
+
+        assertThat(prohibitedController.annotations).anySatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{it.toString() == "SecurityRule.IS_ANONYMOUS"}}
+
+        assertThat(prohibitedController.parameters).noneSatisfy{parameter ->
+            assertThat(parameter.name).isEqualTo("authentication")
+            assertThat(parameter.type.toString()).isEqualTo("io.micronaut.security.authentication.Authentication")}
+
+
+        val requiredController = (controller.find{it.name == "RequiredController"}!!.members.single() as TypeSpec).funSpecs.single()
+
+        assertThat(requiredController.annotations).anySatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}}
+
+        assertThat(requiredController.parameters).anySatisfy{parameter ->
+            assertThat(parameter.name).isEqualTo("authentication")
+            assertThat(parameter.type.toString()).isEqualTo("io.micronaut.security.authentication.Authentication")}
+
+
+        val optionalController = (controller.find{it.name == "OptionalController"}!!.members.single() as TypeSpec).funSpecs.single()
+
+        assertThat(optionalController.annotations).anySatisfy{annotation ->
+            assertThat(annotation.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(annotation.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}}
+
+        assertThat(optionalController.parameters).anySatisfy{parameter ->
+            assertThat(parameter.name).isEqualTo("authentication")
+            assertThat(parameter.type.toString()).isEqualTo("io.micronaut.security.authentication.Authentication?")
```

Comment:
**issue:** Please add the same check to the Spring tests.

Reply:
done